assets/scripts/submit.js

```diff
@@ -1,159 +1,30 @@
   const submitButton = document.getElementById('login-submit')
-  const twoFactorPasscodeInput = document.getElementById('two-factor-passcode')
-  const twoFactorTokenInput = document.getElementById('two-factor-token')
-  const twoFactorTrustDeviceCheckbox = document.getElementById('two-factor-trust-device')
-  const longRunSessionCheckbox = document.getElementById('long-run-session')
-  const twoFactorForms = document.getElementsByClassName('two-factor-form')
-  const passwordForms = document.getElementsByClassName('password-form')
-  const csrfTokenInput = document.getElementById('csrf_token')
-
-  let errorPanel = loginForm && loginForm.querySelector('.wizard-errors')
 
   const twoFactorTrustedDeviceTokenKey = 'two-factor-trusted-device-token'
+  const twoFactorTrustedDomainInput = document.getElementById(
+    'two-factor-trusted-device-token'
+  )
+
   let localStorage = null
   try {
     localStorage = window.localStorage
-  } catch(e) {}
-
-  const showError = function (error) {
-    if (error) {
-      error = '' + error
-    } else {
-      error = 'The Cozy server is unavailable. Do you have network?'
-    }
-
-    if (!errorPanel) {
-      errorPanel = document.createElement('p')
-      errorPanel.classList.add('wizard-errors', 'u-error')
-      loginField.insertBefore(errorPanel, loginField.firstChild);
-    }
-
-    errorPanel.textContent = error;
-    submitButton.removeAttribute('disabled')
-  }
-
-  const onSubmitPassphrase = function(event) {
-    event.preventDefault()
-    submitButton.setAttribute('disabled', true)
-
-    const passphrase = passphraseInput.value
-    const longRunSession = longRunSessionCheckbox.checked ? '1' : '0'
-    const twoFactorTrustedDeviceToken = (localStorage && localStorage.getItem(twoFactorTrustedDeviceTokenKey)) || ''
-    const redirect = redirectInput.value + window.location.hash
-    let headers = new Headers()
-    headers.append('Content-Type', 'application/x-www-form-urlencoded')
-    headers.append('Accept', 'application/json')
-    const reqBody = 'passphrase=' + encodeURIComponent(passphrase) +
-      '&two-factor-trusted-device-token=' + encodeURIComponent(twoFactorTrustedDeviceToken) +
-      '&long-run-session=' + encodeURIComponent(longRunSession) +
-      '&redirect=' + encodeURIComponent(redirect) +
-      '&csrf_token=' + encodeURIComponent(csrfTokenInput.value);
-    fetch('/auth/login', {
-      method: 'POST',
-      headers: headers,
-      body: reqBody,
-      credentials: 'same-origin'
-    }).then(function(response) {
-      const loginSuccess = response.status < 400
-      response.json().then(function(body) {
-        if (loginSuccess) {
-          if (body.two_factor_token) {
-            renderTwoFactorForm(body.two_factor_token)
-            return
-          }
-          submitButton.childNodes[1].innerHTML = '<svg width="16" height="16"><use xlink:href="#fa-check"/></svg>'
-          submitButton.classList.add('c-btn--highlight')
```

Comment:
How did you preserve this animation?

Reply:
It was unfortunately not preserved. I can however re-add it by going back and check the login & redirect using the JS code instead of the html form